### PR TITLE
Waitlist bug fixes and improvements

### PR DESCRIPTION
**Pull request by mateuszkrasucki**
- FIX [backend] when item priority changes move it in queue
    - Items that were not forcefully moved to the top
      should change place on the waitlist as the result of
      priority change.
- FIX disallows adding patient to waitlist mutliple times
  Dialogue displaying details of current entry is displayed
  offering option to edit main complaint.
- UPD default urgent status on adding to waitlist is No
- ADD confirmation popup for adding to waitlist
- FIX various visual problems and conformity with design prototype
- UPD move waitlist handlers from cmd to service  …
- ADD waitlist service
- REM migrateVitalSigns from waitlist backend 
- UPD [waitlist backend] multiple items per patient
- UPD handle in frontend 409 on adding to waitlist 
- Closes: #117, #130, #140, #146

### CODE REVIEW

File: frontend/local/src/containers/waitlist/add/index.js [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
+        this.edit = this.edit.bind(this)
+        this.save = this.save.bind(this)
+        this.close = this.close.bind(this)
+
+        this.state = {
+            edit: false,
+            saving: false,
+            saved: false,
+            adding: false,
+            added: false
```
Comment (dominikznidar) [inline frontend/local/src/containers/waitlist/add/index.js]: this a duplicate of variables already tracked in `wailist.js` module

File: frontend/shared/forms/renderField.js [hunk 2]
```diff
@@ -124,13 +124,13 @@ const renderRadio = ({ input, className, label, options, hideLabel, meta: { touc
                     checked={input.value === option.value}
                     value={option.value}
                 />
-                <label className="form-check-label" htmlFor={`${input.name}${index}`}>
+                <span className="form-check-label" htmlFor={`${input.name}${index}`}>
```
Comment (dominikznidar) [inline frontend/shared/forms/renderField.js]: this should be changed backed to `label` as a lot of people are used to clicking labels to select a radio button

File: frontend/shared/containers/patient/index.js [hunk 3]
```diff
@@ -1,20 +1,20 @@
-import React from "react"
-import classnames from "classnames"
-import moment from "moment"
+import React from "react";
```
Comment (dominikznidar) [inline frontend/shared/containers/patient/index.js]: semicolons

File: frontend/shared/forms/renderField.js [hunk 4]
```diff
@@ -1,9 +1,9 @@
-import React from "react"
-import classnames from "classnames"
-import Select from "react-select"
-import _ from "lodash"
+import React from "react";
```
Comment (dominikznidar) [inline frontend/shared/forms/renderField.js]: semicolons

File: service/waitlist/waitlist.go [hunk 5]
```diff
@@ -126,6 +129,18 @@ func (s *service) DeleteItem(waitlistID, itemID []byte, reason string) error {
...
+
+	// check with the item with provided patientID already exists
+	for i := range currentItems {
+		if *currentItems[i].PatientID == *item.PatientID {
+			return nil, utils.NewError(utils.ErrConflict, "Waitlist already contains item with provided patientID")
+		}
+	}
+
 	return s.storage.AddItem(waitlistID, item)
 }
```
Comment (dominikznidar) [inline service/waitlist/waitlist.go]: This just shortens the window in which the race condition can occur. Should we move this check inside the bolt transaction in the storage?

Comment (mateuszkrasucki) [reply service/waitlist/waitlist.go]: ok

**Pull request merged**